### PR TITLE
UNG-2756 retry endpoints

### PR DESCRIPTION
**Pull request by wojciechUrbanski**

### CODE REVIEW

File: poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala [hunk 1]
```diff
@@ -420,13 +426,63 @@ class DefaultTenantAdminService @Inject() (
...
+        validate(
+          poc.tenantId == tenant.id,
+          ResetPocCreationAttemptsError.NotFound(
+            s"poc: ${poc.id.toString} is not assigned the tenant: ${tenant.id.toString}")
+        )
+      _ <- validate(
+        poc.status == Aborted,
+        ResetPocCreationAttemptsError.PoCNotInAbortedStatus(s"PoC should be in Aborted status but is in ${poc.status}"))
+      allAbortedPocAdmins <- EitherT.right(pocAdminRepository.getAllAbortedByPocId(poc.id))
+      _ <- EitherT.right(pocRepository.updatePoc(poc.copy(status = Processing, creationAttempts = 0)))
```
Comment (Hiroki6) [inline poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala:446]: shall we put transaction here? not so important, though.

File: poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala [hunk 2]
```diff
@@ -420,13 +426,63 @@ class DefaultTenantAdminService @Inject() (
...
+      _ <- allAbortedPocAdmins.traverse_[ResultWithErrorChannel, UUID](pocAdmin =>
```
Comment (Hiroki6) [inline poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala:447]: from performance perspective, this sql is efficient.
more than 10 pocAdmins can belong to one poc.
```sql
update poc_manager.poc_admin_table set status = PROCESSING and creation_attempts = 0 where poc_id = xxx and status = ABORT
```

File: poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala [hunk 3]
```diff
@@ -442,13 +440,11 @@ class DefaultTenantAdminService @Inject() (
...
         ResetPocCreationAttemptsError.PoCNotInAbortedStatus(s"PoC should be in Aborted status but is in ${poc.status}"))
-      allAbortedPocAdmins <- EitherT.right(pocAdminRepository.getAllAbortedByPocId(poc.id))
       _ <- EitherT.right(pocRepository.updatePoc(poc.copy(status = Processing, creationAttempts = 0)))
-      _ <- allAbortedPocAdmins.traverse_[ResultWithErrorChannel, UUID](pocAdmin =>
-        resetPocAdminCreationAttempts[ResetPocCreationAttemptsError](pocAdmin))
+      _ <- EitherT.right(pocAdminRepository.retryAllPocAdmins(poc.id))
     } yield ()
 
-    res.value
+    quillMonixJdbcContext.withTransaction(res.value)
```
Comment (Hiroki6) [inline poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala:447]: why here?

File: poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala (hunk 2 above)
Comment (wojciechUrbanski) [reply poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala]: You are right, fixed ;)

File: poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala (hunk 3 above)
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala:447]: Ah, I got it. nice

**Review by Hiroki6 — APPROVED** [review]

**Pull request merged**